### PR TITLE
Spay neuter zip update

### PR DESCRIPTION
**Pull request by mxsnyder**
The only real changes are to **js/page-specific/spay-neuter-calculator.js**. Everything else is from the recompile.

### CODE REVIEW

File: js/page-specific/spay-neuter-calculator.js [hunk 1]
```diff
@@ -7,379 +7,377 @@ namespacer('baltimoreCounty.pageSpecific');
...
-					setVisibility($spayNeuterFormButton, true, function () {
-						setVisibility($zipCodeField, true, function () {
-							setVisibility($catPitBullField, true, function () {
-								setVisibility($publicAssistanceField, true, function () {
-									calculate();
-								});
+		});
+		$residentField.eq(NO_RADIO_INDEX).on('click', function (event) {
+			clearEverythingOnTheFormAfterThis(event.target);
+			setVisibility($spayNeuterFormResults, true, function () {
```
Comment (martypowell) [inline js/page-specific/spay-neuter-calculator.js]: I wonder if you could refactor this code a bit to pass in an array of elements?

```javascript
setVisibility([$elm1, $elm2'], callback)
```

File: js/page-specific/spay-neuter-calculator.js [hunk 2]
```diff
@@ -7,379 +7,377 @@ namespacer('baltimoreCounty.pageSpecific');
...
-		 */
-		setVisibility = function ($target, shouldHide, callback) {
-			var ANIMATION_DURATION = 300,
-				visibilityFilterSelector = shouldHide ? ':hidden' : ':visible',
-				$formControlWrapper = $target.closest('.bc-form-control');
+	/** 
+	 * Displays the final judgement of the calculator.
+	 */
+	var displayResults = function (discountMessageHTML, facilityListHTML) {
+		$spayNeuterFormResults.removeClass('alert-success alert-warning');
```
Comment (martypowell) [inline js/page-specific/spay-neuter-calculator.js]: Should this be chained?

File: js/page-specific/spay-neuter-calculator.js [hunk 3]
```diff
@@ -7,379 +7,377 @@ namespacer('baltimoreCounty.pageSpecific');
...
-				return '<p>We\'re sorry. Only County residents are eligible for discount spay or neuter procedures.</p>';
+		return facilityArr;
+	};
 
-			if (facilities.length === 1 && facilities[0] === facilityList.dundalk)
-				return '<p>Good news! You\'re eligible for a <strong>free procedure</strong> at our Dundalk facility at 7702 Dunmanway.</p>';
+	/*
+		* Builds the HTML for the discount message.
+		*/
+	var buildDiscountMessageHTML = function (facilities, cost) {
```
Comment (martypowell) [inline js/page-specific/spay-neuter-calculator.js]: should we also check to make sure facilities exist?

File: js/page-specific/spay-neuter-calculator.js [hunk 4]
```diff
@@ -7,379 +7,377 @@ namespacer('baltimoreCounty.pageSpecific');
...
+		* Selects the facility based on cost and ZIP code.
+		*/
+	var facilityPicker = function (cost, isPublicAssistance, isDundalkZip, isSwapZip) {
+		var facilityArr = [];
 
-			if (isDundalkZip) {
-				facilityArr.push(facilityList.dundalk);
-				return facilityArr;
-			}
+		if (typeof cost === 'undefined')
```
Comment (martypowell) [inline js/page-specific/spay-neuter-calculator.js]: Insert brackets for if

File: js/page-specific/spay-neuter-calculator.js [hunk 5]
```diff
@@ -7,379 +7,377 @@ namespacer('baltimoreCounty.pageSpecific');
...
-			var YES_RADIO_INDEX = 0;
+	/*
+		* Pulls Radio Button value safely.
+		*/
+	var getRadioButtonValue = function ($radioButton) {
+		var YES_RADIO_INDEX = 0;
 
-			if ($radioButton.length === 0)
-				return undefined;
+		if ($radioButton.length === 0)
```
Comment (martypowell) [inline js/page-specific/spay-neuter-calculator.js]: Brackets for if

**Comment by martypowell** [comment]
@mxsnyder is this still active?

**Comment by mxsnyder** [comment]
I'm going to close this out since it's almost 3 months old.

**Pull request closed without merging**